src/Components/Components/src/PersistentState/ComponentStatePersistenceManager.cs

```diff
@@ -92,6 +106,25 @@ async Task PersistState(IPersistentComponentStateStore store)
         }
     }
 
+    /// <summary>
+    /// Initializes the render mode for state persisted by the platform.
+    /// </summary>
+    /// <param name="renderMode">The render mode to use for state persisted by the platform.</param>
+    /// <exception cref="InvalidOperationException">when the render mode is already set.</exception>
+    public void SetPlatformRenderMode(IComponentRenderMode renderMode)
+    {
+        if (_servicesRegistry == null)
+        {
+            return;
+        }
+        else if (_servicesRegistry?.RenderMode != null)
+        {
+            throw new InvalidOperationException("Render mode already set.");
+        }
+
+        _servicesRegistry!.RenderMode = renderMode;
+    }
```

Comment:
`Microsoft.AspNetCore.Components` doesn't have access to the common render modes (Server,Webassembly,Auto) so we need to give hosts a way to configure the render mode that is used for persisting the PersistentServicesRegistry